lib/plugins/aws/package/compile/events/mq.js

```diff
@@ -0,0 +1,150 @@
+'use strict';
+
+class AwsCompileMQEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileMQEvents.bind(this),
+    };
+
+    const secretsManagerArnRegex =
+      'arn:[a-z-]+:secretsmanager:[a-z0-9-]+:\\d+:secret:[A-Za-z0-9/_+=.@-]+';
+
+    const mqBrokerArnRegex =
+      'arn:[a-z-]+:mq:[a-z0-9-]+:\\d+:broker:[A-Za-z0-9/_+=.@-]+:b-[a-z0-9-]+';
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'mq', {
+      type: 'object',
+      properties: {
+        arn: {
+          anyOf: [
+            { type: 'string', pattern: mqBrokerArnRegex },
+            { $ref: '#/definitions/awsCfImport' },
+            { $ref: '#/definitions/awsCfRef' },
+          ],
+        },
+        accessConfigurations: {
+          type: 'object',
+          minProperties: 1,
+          maxProperties: 1,
+          properties: {
+            basicAuth: {
+              anyOf: [
+                { type: 'string', pattern: secretsManagerArnRegex },
```

Comment:
If I'm not mistaken, this definition is duplicated across `mq` and `kafka` events - do you think we could move it to schema definitions and reuse it in these two cases?